### DISCUSSION TITLE
The speed of hmmscan

### DISCUSSION THREAD

**Discussion by yl2420841262** [medema-group/BiG-SCAPE#57] [Q&A]
My input file is a gbk file with 394 antismash results, py -i /mnt/f/ NDC-blastp /bigscape -o /mnt/f/ NDC-blastp /results has now been running python bigscape.py -i/MNT /f/ NDC-blastp /results for more than two days, and the code says to start with 394 files
Files that had its sequence extracted: 394

Creating output directories

Trying threading on 32 cores

Predicting domains using hmmscan
Predicting domains for 394 fasta files, I wonder how long it will take

**Comment by jorgecnavarrom** [comment]
That definitely sounds too slow. 

> My input file is a gbk file with 394 antismash results

> Predicting domains for 394 fasta files

Could it be that BiG-SCAPE is processing 394 gbks corresponding to whole-genomes? By default, it will only include files that contain the word 'cluster' or 'region' but that can be changed if you used the `--include-gbk-string`/`--exclude_gbk_str` parameters

**Reply by yl2420841262** [reply]
The 393 gbk files are derived from antismash's analysis and have names similar to NPDC020737.ctg-0001.region001.gbk, 394gbk with a total file size of just 48.6 MB. It now shows that it is still running, but it uses memory instead of cpu

**Reply by yl2420841262** [reply]
![image](https://github.com/medema-group/BiG-SCAPE/assets/138990556/6be61e09-3b4c-4308-a372-ccf49ab79ee3)
